cuda_pathfinder/cuda/pathfinder/_utils/conda_env.py

```diff
@@ -0,0 +1,41 @@
+# SPDX-FileCopyrightText: Copyright (c) 2025 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+import functools
+import os
+from dataclasses import dataclass
+from pathlib import Path
+from typing import Literal, Optional
+
+# https://docs.conda.io/projects/conda-build/en/stable/user-guide/environment-variables.html
+
+BUILD_STATES = ("RENDER", "BUILD", "TEST")
+
+
+@dataclass(frozen=True)
+class CondaPrefix:
+    env_state: Literal["RENDER", "BUILD", "TEST", "activated"]
+    path: Path
+
+
+@functools.cache
+def get_conda_prefix() -> Optional[CondaPrefix]:
+    """
+    Return the effective conda prefix.
+    - RENDER, BUILD, TEST: inside conda-build (host prefix at $PREFIX)
```

Comment:
ditto
```suggestion
    - BUILD, TEST: inside conda-build (host prefix at $PREFIX)
```
also the prefix search is problematic IMHO -- see my comment below.